flash/text/classification/data.py

```diff
@@ -11,247 +11,168 @@
 from flash.core.utilities.imports import _TEXT_AVAILABLE, requires
 from flash.core.utilities.stages import RunningStage
+from flash.text.classification.model import TextClassificationBackboneState
 
 if _TEXT_AVAILABLE:
     from datasets import Dataset, load_dataset
     from transformers import AutoTokenizer, default_data_collator
-    from transformers.modeling_outputs import SequenceClassifierOutput
+else:
+    Dataset = object
 
 
 class TextDeserializer(Deserializer):
     @requires("text")
-    def __init__(self, backbone: str, max_length: int, use_fast: bool = True, **kwargs):
+    def __init__(self, max_length: int):
         super().__init__()
-        self.backbone = backbone
-        self.tokenizer = AutoTokenizer.from_pretrained(backbone, use_fast=use_fast, **kwargs)
         self.max_length = max_length
 
+    @property
+    @lru_cache(maxsize=None)
+    def tokenizer(self):
+        backbone_state = self.get_state(TextClassificationBackboneState)
+        return AutoTokenizer.from_pretrained(backbone_state.backbone, use_fast=True)
+
     def serve_load_sample(self, text: str) -> Tensor:
         return self.tokenizer(text, max_length=self.max_length, truncation=True, padding="max_length")
 
     @property
     def example_input(self) -> str:
         return "An example input"
 
-    def __getstate__(self):  # TODO: Find out why this is being pickled
-        state = self.__dict__.copy()
-        state.pop("tokenizer")
-        return state
-
-    def __setstate__(self, state):
-        self.__dict__.update(state)
-        self.tokenizer = AutoTokenizer.from_pretrained(self.backbone, use_fast=True)
-
-
-class TextInput(Input):
-    @requires("text")
-    def __init__(self, backbone: str, max_length: int = 128):
-        super().__init__()
-
-        self.backbone = backbone
-        self.tokenizer = AutoTokenizer.from_pretrained(backbone, use_fast=True)
-        self.max_length = max_length
-
-    def _tokenize_fn(
-        self,
-        ex: Union[Dict[str, str], str],
-        input: Optional[str] = None,
-    ) -> Callable:
-        """This function is used to tokenize sentences using the provided tokenizer."""
-        return self.tokenizer(ex[input], max_length=self.max_length, truncation=True, padding="max_length")
-
-    @staticmethod
-    def _transform_label(label_to_class_mapping: Dict[str, int], target: str, ex: Dict[str, Union[int, str]]):
-        ex[target] = label_to_class_mapping[ex[target]]
-        return ex
 
+class TextClassificationInput(ClassificationInput):
     @staticmethod
-    def _multilabel_target(targets: List[str], element: Dict[str, Any]) -> Dict[str, Any]:
-        targets = [element.pop(target) for target in targets]
-        element[DataKeys.TARGET] = targets
+    def _resolve_target(target_keys: Union[str, List[str]], element: Dict[str, Any]) -> Dict[str, Any]:
+        if not isinstance(target_keys, List):
+            element[DataKeys.TARGET] = element.pop(target_keys)
```

Comment:
Would this work as expected ?

Reply:
This is just a preprocessing step before our own target handling mechanism. So it doesn't matter what comes out here as we will later handle it properly with our classification utilities.